internal/metadata/metadata.go

```diff
@@ -72,3 +75,41 @@ func Set(addr resolver.Address, md metadata.MD) resolver.Address {
 	addr.Attributes = addr.Attributes.WithValue(mdKey, mdValue(md))
 	return addr
 }
+
+// Validate returns an error if the input md contains invalid keys or values.
+// There are check items:
+// - header names contain one or more characters from this set [0-9 a-z _ - .]
```

Comment:
Did you update this godoc?

Reply:
I don't get it. I did `go doc metadata.Validate` then printing the following message.
```sh
❯ go doc metadata.Validate
package metadata // import "google.golang.org/grpc/internal/metadata"

func Validate(md metadata.MD) error
    Validate returns an error if the input md contains invalid keys or values.

    There are check items: - header names contain one or more characters from
    this set [0-9 a-z _ - .] - if the header-name ends with a "-bin" suffix, the
    header-value could contain an arbitrary octet sequence. So no real
    validation required here. - if header-name does not end with a "-bin"
    suffix, header-value should only contain one or more characters from the set
    ( %x20-%x7E ) which includes space and printable ASCII.
```